src/main/java/io/stargate/sgv2/jsonapi/util/JsonUtil.java

```diff
@@ -58,4 +63,47 @@ public static boolean equalsOrdered(JsonNode node1, JsonNode node2) {
     // For other nodes default equals() works fine:
     return Objects.equals(node1, node2);
   }
+
+  /**
+   * Helper method that may be used to see if given JSON Value looks like it is an EJSON-encoded
+   * value -- that is, JSON Object with 1 entry, name of which starts with "$" character -- without
+   * decoding value. No validation is done beyond feasibility: caller is likely to want to call
+   * other methods like {@link #extractEJsonDate} for further processing and validation/
+   *
+   * @param json JSON value to check
+   * @return {@code true} if value looks like it has to be EJSON value (valid or invalid); {@code
+   *     false} if not.
+   */
+  public static boolean looksLikeEJsonValue(JsonNode json) {
+    return json.isObject() && json.size() == 1 && json.fieldNames().next().startsWith("$");
+  }
+
+  /**
+   * Helper method that will see if given {@link JsonNode} is an EJSON-encoded "Date" (aka
+   * Timestamp) value and if so, constructs and returns matching {@link java.util.Date} value. See
+   * {@href https://docs.meteor.com/api/ejson.html} for details on encoded value.
+   *
+   * @param json JSON value to check
+   * @return
+   */
+  public static Date extractEJsonDate(JsonNode json, Object path) {
+    if (json.isObject() && json.size() == 1) {
+      JsonNode value = json.get(EJSON_VALUE_KEY_DATE);
+      if (value != null) {
+        if (value.isIntegralNumber() && value.canConvertToLong()) {
+          return new Date(value.longValue());
```

Comment:
Do we need to to wrap this into a date or instant at all?

Reply:
Not on server side; we just want something to hold on to that wraps a `long` and is type-safe for documentation.
`Instant` has 2 `longs` (millisecond like `Date`, and then nanosecond counter).

Mongo has only one date-related operator (`$currentDate`), for everything else we just convert to/from JSON, or get 64-bit value (for sorting).